### DISCUSSION TITLE
Does it really make sense to have a dimensionality for numeric data types?

### DISCUSSION THREAD

**Discussion by neumannjan** [LukasZahradnik/deep-db-learning#17] [General]
Currently the numeric convertor contains trainable weights by which the numerical scalar value gets scaled.

However, such weights can currently have an arbitrary dimension. Does that make sense?

If the input is simply a scalar, shouldn't the output be the same scalar, just scaled by a (scalar) trainable weight?

**Comment by LukasZahradnik** [comment]
It is supposed to be processed by a transformer (as a token), and we need all tokens to be of the same dimensionality. Also, I'm not sure whether we can have jagged tensors as node features.
